File: stepik-java-api/src/main/java/org/stepik/api/client/TransportClient.java
package org.stepik.api.client;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Map;

/**
 * @author meanmail
 */
public interface TransportClient {
    @NotNull
    ClientResponse post(@NotNull StepikApiClient stepikApiClient, @NotNull String url, @Nullable String body);

    @NotNull
    ClientResponse get(@NotNull StepikApiClient stepikApiClient, @NotNull String url);

    @NotNull
    ClientResponse post(
            @NotNull StepikApiClient stepikApiClient,
            @NotNull String url,
            @Nullable String body,
            @Nullable Map<String, String> headers);

    @NotNull
    ClientResponse get(
            @NotNull StepikApiClient stepikApiClient,
            @NotNull String url,
            @Nullable Map<String, String> headers);
}

File: stepik-java-api/src/main/java/org/stepik/api/client/TransportClient.kt
package org.stepik.api.client


interface TransportClient {
    fun post(stepikApiClient: StepikApiClient, url: String, body: String?): ClientResponse

    operator fun get(stepikApiClient: StepikApiClient, url: String): ClientResponse

    fun post(
            stepikApiClient: StepikApiClient,
            url: String,
            body: String?,
            headers: Map<String, String>?): ClientResponse

    operator fun get(
            stepikApiClient: StepikApiClient,
            url: String,
            headers: Map<String, String>?): ClientResponse
}
